src/urllib3/_request_methods.py

```diff
@@ -55,7 +55,7 @@ def urlopen(
         encode_multipart: bool = True,
         multipart_boundary: Optional[str] = None,
         **kw: Any,
-    ) -> BaseHTTPResponse:  # Abstract
+    ) -> HTTPResponse:  # Abstract
```

Comment:
We don't want to have too specific a type here, returning `BaseHTTPResponse` is intentional.

Reply:
Perhaps `RequestMethods` can be made generic in the response type, with the type var bound by `BaseHTTPResponse`. Then each class can extend with its actual response type? (I didn't check if that makes sense)

Aside: I personally dislike mixin classes like `RequestMethods`, but that's another discussion...

Reply:
(RequestMethods is definitely confusing.)

Reply:
@sethmlarson I've reverted my commit for changing the type.

@bluetech
> Perhaps RequestMethods can be made generic in the response type, with the type var bound by BaseHTTPResponse. Then each class can extend with its actual response type? 

Could you please give me an example or explain how to do it?

Reply:
Change `RequestMethods` to

```py
BaseHTTPResponseT = TypeVar('BaseHTTPResponseT', bound=BaseHTTPResponse)

class RequestMethods(Generic[BaseHTTPResponseT]):
    def request(...) -> BaseHTTPResponseT: ...
```

and then for each mixing-in inherit from `RequestMethods[TheActualBaseHTTPResponseTypeForThisClass]`.

Again, didn't check if it would actually help, but it allows each class to decide which response type it wants for the mixed-in methods.

Reply:
Thanks @bluetech for your code example. I've changed it as you suggested